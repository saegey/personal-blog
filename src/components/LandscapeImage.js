/** @jsxImportSource theme-ui */
import * as React from "react"
import { Image, Box } from "theme-ui"
import { getImage } from "gatsby-plugin-image"
import { GatsbyImage } from "gatsby-plugin-image"

import ImageWrapper from "./imageWrapper.js"
import FullScreenIcon from "./FullScreenIcon"

const LandscapeImage = ({ image, caption }) => {
  return (
    <ImageWrapper image={image} caption={caption}>
      <Box sx={{ width: "100%", position: "relative" }}>
        <Image
          image={getImage(image)}
          alt={"blah"}
          as={GatsbyImage}
          sx={{
            width: ["100%", "100%", "100%"],
            zIndex: "100",
            borderRadius: [4, 4, 4],
            // backdropFilter: "blur(10px)",
          }}
        ></Image>
        <Box
          sx={{
            width: "40px",
            position: "absolute",
            right: "0",
            top: "0",
            zIndex: 10000,
          }}
        >
          <FullScreenIcon />
        </Box>
      </Box>
    </ImageWrapper>
  )
}

export default LandscapeImage
