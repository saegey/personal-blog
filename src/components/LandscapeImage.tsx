import { Box, useColorMode } from 'theme-ui'
import { getImage, IGatsbyImageData } from 'gatsby-plugin-image'

import ImageWrapper from './imageWrapper'
import FullScreenIcon from './FullScreenIcon'
import CustomImage from './CustomImage'

interface RaceImageType {
  image: IGatsbyImageData
  caption: string
  altText: string
  invert?: boolean
  maximize?: boolean
  roundedEdges?: boolean
}

const LandscapeImage = ({
  image,
  caption,
  invert = false,
  maximize = true,
  roundedEdges = true,
}: RaceImageType) => {
  const [colorMode] = useColorMode()

  if (!maximize) {
    return (
      <Box sx={{ width: '100%', position: 'relative' }}>
        <CustomImage
          image={getImage(image)}
          alt={'blah'}
          theme={{
            filter: invert && colorMode === 'dark' ? `invert(1)` : `none`,
            width: ['100%', '100%', '100%'],
            borderRadius: roundedEdges ? [5, 5, 5] : [0, 0, 0],
            '-webkit-mask-image': '-webkit-radial-gradient(white, black)',
          }}
        />
      </Box>
    )
  }

  return (
    <ImageWrapper image={image} caption={caption} altText="">
      <Box sx={{ width: '100%', position: 'relative' }}>
        <CustomImage
          image={getImage(image)}
          alt={'blah'}
          theme={{
            filter: invert && colorMode === 'dark' ? `invert(1)` : `none`,
            width: ['100%', '100%', '100%'],
            borderRadius: [4, 4, 4],
            // '-webkit-mask-image': '-webkit-radial-gradient(white, black)',
          }}
        />
        <Box
          sx={{
            width: '40px',
            position: 'absolute',
            right: '0',
            top: '0',
            zIndex: 0,
          }}
        >
          <FullScreenIcon />
        </Box>
      </Box>
    </ImageWrapper>
  )
}

export default LandscapeImage
