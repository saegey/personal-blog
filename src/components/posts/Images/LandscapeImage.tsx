import { Box, useColorMode, useThemeUI } from 'theme-ui'
import { getImage, IGatsbyImageData } from 'gatsby-plugin-image'

import ImageWrapper from './imageWrapper'
import FullScreenIcon from '../../icons/FullScreenIcon'
import CustomImage from '../../CustomImage'

interface RaceImageType {
  image: IGatsbyImageData
  caption: string
  altText: string
  invert?: boolean
  maximize?: boolean
  roundedEdges?: boolean
}

const LandscapeImage = ({
  image,
  caption,
  invert = false
}: RaceImageType) => {
  const [colorMode] = useColorMode()
  const { theme } = useThemeUI()

  return (
    <ImageWrapper image={image} caption={caption} altText="">
      <Box sx={{ width: '100%', position: 'relative' }}>
        <CustomImage
          image={getImage(image)}
          alt={'blah'}
          theme={{
            filter: invert && colorMode === 'dark' ? `invert(1)` : `none`,
            width: ['100%', '100%', '100%'],
            borderRadius: [4, 4, 4],
            WebkitMaskImage: '-webkit-radial-gradient(white, black)',
          }}
        />
        <Box
          sx={{
            height: '32px',
            width: '32px',
            padding: '2px',
            position: 'absolute',
            right: '10px',
            top: '10px',
            zIndex: 0,
          }}
        >
          <FullScreenIcon color={String(theme.colors?.background)} />
        </Box>
      </Box>
    </ImageWrapper>
  )
}

export default LandscapeImage
