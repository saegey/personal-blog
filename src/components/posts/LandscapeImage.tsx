import { Box, useColorMode } from 'theme-ui'
import { getImage, IGatsbyImageData } from 'gatsby-plugin-image'

import ImageWrapper from '../imageWrapper'
import FullScreenIcon from '../icons/FullScreenIcon'
import CustomImage from '../CustomImage'

interface RaceImageType {
  image: IGatsbyImageData
  caption: string
  altText: string
  invert?: boolean
  maximize?: boolean
  roundedEdges?: boolean
}

const LandscapeImage = ({
  image,
  caption,
  invert = false,
  maximize = true,
  roundedEdges = true,
}: RaceImageType) => {
  const [colorMode] = useColorMode()

  if (!maximize) {
    return (
      <Box sx={{ width: '100%', position: 'relative' }}>
        <CustomImage
          image={getImage(image)}
          alt={'blah'}
          theme={{
            filter: invert && colorMode === 'dark' ? `invert(1)` : `none`,
            width: ['100%', '100%', '100%'],
            borderRadius: roundedEdges ? [5, 5, 5] : [0, 0, 0],
            '-webkit-mask-image': '-webkit-radial-gradient(white, black)',
          }}
        />
      </Box>
    )
  }

  return (
    <ImageWrapper image={image} caption={caption} altText="">
      <Box sx={{ width: '100%', position: 'relative' }}>
        <CustomImage
          image={getImage(image)}
          alt={'blah'}
          theme={{
            filter: invert && colorMode === 'dark' ? `invert(1)` : `none`,
            width: ['100%', '100%', '100%'],
            borderRadius: [4, 4, 4],
            // '-webkit-mask-image': '-webkit-radial-gradient(white, black)',
          }}
        />
        <Box
          sx={{
            height: '32px',
            width: '32px',
            padding: '2px',
            position: 'absolute',
            right: '10px',
            top: '10px',
            zIndex: 0,
          }}
        >
          <FullScreenIcon />
        </Box>
      </Box>
    </ImageWrapper>
  )
}

export default LandscapeImage
