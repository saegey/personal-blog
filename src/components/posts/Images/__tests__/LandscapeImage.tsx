/**
 * @jest-environment jsdom
 */
import renderer from 'react-test-renderer'

import LandscapeImage from '../LandscapeImage'

jest.mock('theme-ui', () => {
  return {
    Text: ({ children }: { children: JSX.Element }) => {
      return <p>{children}</p>
    },
    Box: ({ children }: { children: JSX.Element }) => {
      return <div>{children}</div>
    },
    Image: ({ children }: { children: JSX.Element }) => {
      return <img>{children}</img>
    },
    useColorMode: () => {
      return ['dark']
    },
    useThemeUI: () => {
      return {
        theme: {
          colors: {
            background: 'blue',
          },
        },
      }
    },
  }
})

describe('LandscapeImage', () => {
  it('renders', () => {
    const json = renderer
      .create(
        <LandscapeImage
          image={{
            layout: 'constrained',
            width: 100,
            height: 200,
            images: {
              fallback: {
                src: 'http://fsaf.com/1.jpg',
              },
              sources: [{ media: 'balf', type: 'fla', srcSet: 'blah' }],
              // altText: 'alt teext',
              // shouldLoad: true,
            },

            backgroundColor: 'black',
          }}
          caption="this is a caption"
          maximize={false}
        />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
