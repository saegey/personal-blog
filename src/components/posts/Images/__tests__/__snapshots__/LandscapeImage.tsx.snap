// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandscapeImage renders 1`] = `
<div>
  <div>
    <div>
      <img />
      <div>
        <div>
          <g
            fill="none"
            fillRule="evenodd"
            id="Page-1"
            stroke="none"
            strokeWidth="1"
          >
            <g
              className="icon-solid"
              id="Core"
              transform="translate(-215.000000, -257.000000)"
            >
              <g
                id="fullscreen"
                transform="translate(215.000000, 257.000000)"
              >
                <path
                  d="M2,9 L0,9 L0,14 L5,14 L5,12 L2,12 L2,9 L2,9 Z M0,5 L2,5 L2,2 L5,2 L5,0 L0,0 L0,5 L0,5 Z M12,12 L9,12 L9,14 L14,14 L14,9 L12,9 L12,12 L12,12 Z M9,0 L9,2 L12,2 L12,5 L14,5 L14,0 L9,0 L9,0 Z"
                  id="Shape"
                />
              </g>
            </g>
          </g>
        </div>
      </div>
    </div>
  </div>
  <p>
    this is a caption
  </p>
</div>
`;
